### PR TITLE
techDebt: [PL-24942]: Added error handling in case of fetch Secrets

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35081)

### CODE REVIEW

File: 890-sm-core/src/main/java/io/harness/encryptors/clients/AwsSecretsManagerEncryptor.java [hunk 1]
```diff
@@ -96,8 +98,16 @@ public EncryptedRecord createSecret(
         failedAttempts++;
         log.warn("encryption failed. trial num: {}", failedAttempts, e);
         if (failedAttempts == NUM_OF_RETRIES) {
```
Comment (shreyas-harness) [inline 890-sm-core/src/main/java/io/harness/encryptors/clients/AwsSecretsManagerEncryptor.java:100]: Rather than having while(true) and then have failedAttempts == NUM_OF_RETRIES we should have, 
for ( int failedAttempts = 0; failedAttemps<=NUM_OF_RETRIES; failedAttemps++) { 
    try{ 
...
}
catch(Exception e) {
if(failedAttemps==NUM_OF_RETRIES){
... // based on exception
throw (ex)
}


Also note that sleep statement in catch block can itself throw an exception

**Review by shreyas-harness — COMMENTED** [review]
Same comment for all the 4 methods

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35081)* status: Some of the checks are too old. 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`890-sm-core/src/test/java/io/harness/encryptors/AwsSecretsManagerEncryptorTest.java` line 381 at r2](https://reviewable.io/reviews/harness/harness-core/35081#-N87R9C-3BkTSbmoLr76:-N87R9C-3BkTSbmoLr77:b3cx6dv) ([raw file](https://github.com/harness/harness-core/blob/e655f8edfb0e45daa93fb681eea3968d0db4dcdf/890-sm-core/src/test/java/io/harness/encryptors/AwsSecretsManagerEncryptorTest.java#L381)):*
> ```Java
>           awsSecretsManagerConfig.getAccountId(), secretName, plainTextValue, awsSecretsManagerConfig);
>       fail("Create Secret should have failed");
>     } catch (AWSSecretsManagerException e) {
> ```

add test case for UncheckedTimeoutException and case where none of the specified exception caught. Same for other methods i.e. update and fetch.

**Review by VikasMaddukuriHarness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35081)* status: 2 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`890-sm-core/src/test/java/io/harness/encryptors/AwsSecretsManagerEncryptorTest.java` line 381 at r2](https://reviewable.io/reviews/harness/harness-core/35081#-N87R9C-3BkTSbmoLr76:-N88Wx-7CHbd3KMqTYBR:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7fbb28d42450e63647fb5526774d7a5c4ae1ad8f/890-sm-core/src/test/java/io/harness/encryptors/AwsSecretsManagerEncryptorTest.java#L381)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

add test case for UncheckedTimeoutException and case where none of the specified exception caught. Same for other methods i.e. update and fetch.
</blockquote></details>

Done.



_Code quote:_
```Java

  @Test
  @Owner(developers = UTKARSH)
  @Category(UnitTests.class)
```

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35081)* status: 1. unresolved discussions.

**Review by shreyas-harness — COMMENTED** [review]
> Reviewed 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35081)* status: 1. unresolved discussions. 0 of 1 approvals obtained.

**Review by shreyas-harness — DISMISSED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35081#-:-N88aM4BApM5jFk5hUHe:bnfp4nl)* status: 1. unresolved discussions.

**Review by shreyas-harness — COMMENTED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35081)* status: 1. unresolved discussions. 0 of 1 approvals obtained.

**Review by shreyas-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35081)* status: :shipit: complete! ✓

**Review by shreyas-harness — DISMISSED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35081#-:-N88e22n4zAi4uXbfW5r:bnfp4nl)* status: :shipit: complete! ✓

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35081)* status: :shipit: complete! ✓
